clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/DefaultIngestionBackend.java

```diff
@@ -115,6 +124,51 @@ CompletionStage<Void> bootstrapFromBlobs(Store store, int versionNumber, int par
     });
   }
 
+  /**
+   * A helper method to check if the offset lag is within the allowed threshold.
+   * If the offset lag is smaller than the `blobTransferDisabledOffsetLagThreshold`,
+   * bootstrapping from Kafka firstly, even if blob transfer is enabled.
+   *
+   * @param store the store name
+   * @param versionNumber the version number
+   * @param partition the partition number
+   * @param blobTransferDisabledOffsetLagThreshold the maximum allowed offset lag threshold.
+   *        If the offset lag is within this threshold, bootstrapping from Kafka is allowed, even if blob transfer is enabled.
+   *        If the lag exceeds this threshold, bootstrapping should happen from blobs transfer firstly.
+   *
+   * @return true if the offset lag exceeds the threshold or if the lag is 0, indicating bootstrapping should happen from blobs transfer.
+   *         false otherwise
+   */
+  public boolean isOffsetLagged(
+      String store,
+      int versionNumber,
+      int partition,
+      long blobTransferDisabledOffsetLagThreshold) {
+    String topicName = Version.composeKafkaTopic(store, versionNumber);
+    OffsetRecord offsetRecord = storageMetadataService.getLastOffset(topicName, partition);
+    LOGGER.info(
+        "Store {} partition {} topic offset is {}, offset lag is {}",
+        store,
+        partition,
+        offsetRecord.getLocalVersionTopicOffset(),
+        offsetRecord.getOffsetLag());
+
+    if (offsetRecord == null || (offsetRecord.getOffsetLag() == 0 && offsetRecord.getLocalVersionTopicOffset() == -1)) {
+      return true;
+    }
+
+    if (offsetRecord.getOffsetLag() < blobTransferDisabledOffsetLagThreshold) {
```

Comment:
I am a bit hesitant between HB lag vs offset lag, though right now I think maybe offset lag can also work for DVC as they all ingest from VT instead of RT. HB can be more accurate about how stale we are for a given replica, but we can't get the information unless we make the change to persist it in the OffsetRecord. 

Maybe it is good for now, and let's make this configurable so we can tune it easily in client app.